src/YesSql.Core/Sql/SchemaBuilder.cs

```diff
@@ -84,7 +89,7 @@ public ISchemaBuilder CreateReduceIndexTable(Type indexType, Action<ICreateTable
                 var documentTable = TableNameConvention.GetDocumentTable(collection);
 
                 createTable
-                    .Column<int>("Id", column => column.Identity().NotNull())
+                    .Column<int>("Id", column => column.PrimaryKey().Identity().NotNull())
```

Comment:
Yes, i was looking at the code to see if here also you don't need an index on the `Id`, but was not sure because here you were not using `PrimaryKey()`, but then was still blocked on this line in `BaseCommandInterpreter .cs`

    var primaryKeys = command.TableCommands.OfType<CreateColumnCommand>()
        .Where(ccc => ccc.IsPrimaryKey && !ccc.IsIdentity).Select(ccc =>
            _dialect.QuoteForColumnName(ccc.ColumnName)).ToArray();

Where it seems that to be a primary key it needs to be `IsPrimaryKey` but not `IsIdentity`, that said looking in a sql table i could see them as primary keys.

Reply:
Maybe in some dbs Indentity is a synonym of PK ... I will try to understand why it's filtering our Identity columns.
If they are not marked as PKs, then there would be no index.

Reply:
Found it! The `IdentityColumnString` sql in the dialect will also set it as PK. That's why the PK statement is only needed to PKs that are not Identity. Adding some comments.

Reply:
Okay cool !!!

Reply:
So, would just using  `.Identity()` have been enough?

Reply:
Okay, just saw that you just updated it ;)